File: code/NutritionTracker/src/main/java/edu/westga/cs3212/group5/nutritiontracker/model/FoodItem.java
package edu.westga.cs3212.group5.nutritiontracker.model;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;

@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
@JsonSubTypes({ @JsonSubTypes.Type(value = CompositeFood.class, name = "composite"),
		@JsonSubTypes.Type(value = BaseFood.class, name = "base") })

/**
 * The Interface FoodItem.
 */
public interface FoodItem {
	public static final String FOOD_ITEMS_JSON_FILE = "food_items.json";
	
	String getDescription();

	void setDescription(String description);
	
	QuantityCategory getQuantityCategory();
	
	void setQuantityCategory(QuantityCategory quantityCategory);

	double getPortionSize();
	
	void setPortionSize(double portionSize);

	double getCalories();

	double getProtein();

	double getFat();

	double getSugar();

	double getCarbohydrates();

	double getSodium();
}

File: code/Group5NutritionServer/src/model/food_item.py
'''
Created on Mar 22, 2026

@author: Justin Smith
'''

from __future__ import annotations

from abc import ABC, abstractmethod


class FoodItem(ABC):
    @abstractmethod
    def get_description(self):
        pass

    @abstractmethod
    def get_quantity_category(self):
        pass

    @abstractmethod
    def get_portion_size(self):
        pass

    @abstractmethod
    def get_calories(self):
        pass

    @abstractmethod
    def get_protein(self):
        pass

    @abstractmethod
    def get_fat(self):
        pass

    @abstractmethod
    def get_sugar(self):
        pass

    @abstractmethod
    def get_carbohydrates(self):
        pass

    @abstractmethod
    def get_sodium(self):
        pass